### PR TITLE
Run deploy openstack bundle

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Add run_deploy_lxd_profile_bundle to integration tests, replacing nw-deploy-lxd-profile-bundle-lxd-openstack once stable.  

Add idle_subordinate_condition() as a condition for use in wait_for

## QA steps

```sh
(cd tests ; ./main.sh deploy test_deploy_bundles)
```

### CODE REVIEW

**Comment by manadart** [comment]
Sorry, I jumped the gun before my run actually got to this test. It gets stuck with:
`grep: Invalid range end`

**Comment by hmlanigan** [comment]
@manadart, the issues was caused by a lack of space between the first 2 args of wait_for here: https://github.com/juju/juju/pull/11110/files#diff-395efcca571ec5165ed88d2d1adb7bd5R94 .  The value of the name variable changes after call to idle_condition.

+ idle_condition rabbitmq-server 9
+ local name app_index unit_index
**+ name=rabbitmq-server**
+ app_index=9
+ unit_index=0
+ echo .applications | select(.["rabbitmq-server"] | .units | .["rabbitmq-server/0"] | .["juju-status"] | .current == "idle") | keys[9]
+ wait_for rabbitmq-server.applications | select(.["rabbitmq-server"] | .units | .["rabbitmq-server/0"] | .["juju-status"] | .current == "idle") | keys[9]    <- lost a space between “rabbitmq-server” and “.applications”
+ local name query
**+ name=rabbitmq-server.applications | select(.["rabbitmq-server"] | .units | .["rabbitmq-server/0"] | .["juju-status"] | .current == "idle") | keys[9]**

Pushing up the fix shortly.

File: tests/includes/wait-for.sh [hunk 1]
```diff
@@ -18,14 +18,14 @@ wait_for() {
     attempt=0
     # shellcheck disable=SC2046,SC2143
     until [ "$(juju status --format=json 2> /dev/null | jq -S "${query}" | grep "${name}")" ]; do
-        echo "[+] (attempt ${attempt}) polling status"
+        echo "[+] (attempt ${attempt}) polling status for ${name}"
```
Comment (SimonRichardson) [inline tests/includes/wait-for.sh]:
```suggestion
        echo "[+] (attempt ${attempt}) polling status for "${name}""
```

File: tests/includes/wait-for.sh [hunk 2]
```diff
@@ -18,14 +18,14 @@ wait_for() {
...
         juju status --relations 2>&1 | sed 's/^/    | /g'
         sleep "${SHORT_TIMEOUT}"
         attempt=$((attempt+1))
     done
 
     if [ "${attempt}" -gt 0 ]; then
-        echo "[+] $(green 'Completed polling status')"
+        echo "[+] $(green 'Completed polling status for') $(green ${name})"
```
Comment (SimonRichardson) [inline tests/includes/wait-for.sh]:
```suggestion
        echo "[+] $(green 'Completed polling status for') $(green "${name}")"
```

File: tests/includes/wait-for.sh (hunk 1 above)
Comment (hmlanigan) [reply tests/includes/wait-for.sh]: done

File: tests/includes/wait-for.sh (hunk 2 above)
Comment (hmlanigan) [reply tests/includes/wait-for.sh]: done

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
intermittent failure in worker/caasoperator for merge job, check job was good.

$$merge$$

**Pull request merged**